### PR TITLE
No need to return VIEWs, only TABLEs

### PR DESCRIPTION
**Pull request by jadb**
@see http://stackoverflow.com/questions/2908680/how-to-get-only-tables-not-views-using-show-tables

### CODE REVIEW

File: src/Database/Schema/MysqlSchema.php [hunk 1]
```diff
@@ -28,7 +28,10 @@ class MysqlSchema extends BaseSchema
      */
     public function listTablesSql($config)
     {
-        return ['SHOW TABLES FROM ' . $this->_driver->quoteIdentifier($config['database']), []];
+        return [
+            'SHOW FULL TABLES FROM ' . $this->_driver->quoteIdentifier($config['database']) . ' WHERE Table_Type != ?',
+            ['VIEW']
```
Comment (markstory) [inline src/Database/Schema/MysqlSchema.php]: Why don't we want views? There have been a bunch of recent issues specifically requesting we better support views in SQLServer.

**Comment by markstory** [comment]
SQLServer issue #6501

**Comment by jadb** [comment]
Maybe we could offer a `listViews()`?

**Comment by markstory** [comment]
They could be separate, but then bake would need to union the view + table results.

**Comment by jadb** [comment]
I'll wait and see what others think of that first.

**Comment by jippi** [comment]
-1

Bownty uses quite a lot of views in our db, and we would expect them to be exposed as any other table at all times.

Its very common for big companies and big databases to drop views in the db to hide the complex join rules etc and easy app development

**Comment by dereuromark** [comment]
Maybe it would be better to have a $config option key `type` that would default to null (all).
This way you can opt in `type` `Table/View` and subselect this way without changing the default behavior.

**Comment by jhli** [comment]
I do like those SQL views too, Jippi pointed out the reasons well.

**Comment by jadb** [comment]
@jippi you seem to have missed the suggestion ;)

I am not trying to stop showing VIEWs, but giving the option of showing VIEWs or FULL TABLEs (as those are 2 different things). So, in other words, big companies can still use VIEWs (I don't think its just big companies that do though :wink:)

So please, just because it seemed to have stirred the direction in a different way, the suggestion is to have a `listViews()` and `listTables()` - maybe the difference between the 2 could be the `listFullTables()`.

**Comment by ionas** [comment]
:+1: @dereuromark then.

**Comment by markstory** [comment]
Why do we want to separate the tables and views in the first place? They are functionally the same for many applications.

**Comment by jadb** [comment]
I ran into the problem when trying to loop through tables and drop them (views aren't treated as tables in that case). In any case, if you guys don't see an advantage to being able to get views and tables separately (while keeping the default behaviour), feel free to close it :)

**Comment by jhli** [comment]
Well another diff is that usually they are read only/not materialized. Internally if treated equal there should be table, materialized view and view?

**Comment by jippi** [comment]
@jadb as long nothing changes in the user-land, I don't care :)

Auto-Tables etc should continue to work as always, also considering https://dev.mysql.com/doc/refman/5.6/en/view-updatability.html it can be exposed as a normal table

Maybe have it as a property in the Table class if the instance is referencing a table or view ?

**Comment by jadb** [comment]
@jippi I must admit I like the simplicity of your suggested workaround. And yes, I agree that it shouldn't change current default behaviour.

As stated, the issue is happening not as part as code I would write but when trying to use bake and/or migrations. Migrations would fail (on DROP) and bake would fail at generating a fixture. These are the bugs that led me to think it is best resolved at the source, but I could be wrong.

**Comment by jhli** [comment]
I think rather than dropping VIEW support there should be support fir materialized and read only views across ORM, migrations and bake?

**Comment by jhli** [comment]
And that property Jippi named should be set to one constant of three and dereuromarks siggestion could be implemented too

**Comment by markstory** [comment]
I think having a listViews() method on the schmeacollection class makes sense. Having more indepth support for views is totally an option as well. Being able to create them with fixtures would be awesome.

My concern with the current change is that it could break apps that depend on views right now. Until we have a more holistic solution for views, I don't think we shoukd change the semantics of listTables().

**Comment by markstory** [comment]
Closing as this will certainly break some existing applications. If we want better ways to reflect only views, or only tables I think different code will be needed anyways.

**Pull request closed without merging**